### PR TITLE
enhance: make balance cost threshold configurable

### PR DESCRIPTION
**Pull request by sunby**

### CODE REVIEW

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -1596,6 +1597,16 @@ func (p *queryCoordConfig) init(base *BaseTable) {
 	}
 	p.GrowingRowCountWeight.Init(base.mgr)
 
+	p.BalanceCostThreshold = ParamItem{
+		Key:          "queryCoord.balanceCostThreshold",
+		Version:      "2.3.5",
```
Comment (congqixia) [inline pkg/util/paramtable/component_param.go:1602]: 2.4.0?

File: internal/querycoordv2/balance/multi_target_balance.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (m *segmentCountCostModel) cost() float64 {
 }
 
 func cmpFloat(f1, f2 float64) int {
```
Comment (congqixia) [inline internal/querycoordv2/balance/multi_target_balance.go:98]: NIT: naming this function to `cmpCost`?

File: pkg/util/paramtable/component_param.go [hunk 3]
```diff
@@ -1596,6 +1597,16 @@ func (p *queryCoordConfig) init(base *BaseTable) {
...
+		DefaultValue: "0.001",
+		PanicIfEmpty: true,
+		Doc:          "the threshold of balance cost",
```
Comment (congqixia) [inline pkg/util/paramtable/component_param.go:1605]: NIT: add more comment for this param item meaning later?

File: internal/querycoordv2/balance/multi_target_balance.go (hunk 2 above)
Comment (sunby) [reply internal/querycoordv2/balance/multi_target_balance.go:98]: next pr will change it.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**